<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to mixt ‚Ä¢ mixtR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to mixt">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mixtR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction-to-mixt.html">Introduction to mixt</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vdumeaux/mixt">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to mixt</h1>
                        <h4 class="author">Vanessa Dumeaux</h4>
            
            <h4 class="date">2018-05-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vdumeaux/mixt/blob/master/vignettes/Introduction-to-mixt.Rmd"><code>vignettes/Introduction-to-mixt.Rmd</code></a></small>
      <div class="hidden name"><code>Introduction-to-mixt.Rmd</code></div>

    </div>

    
    
<p>The mixtR package (Matched Interactions across Tissues) contains the computational and statistical methods for identifying and exploring associations between sets of genes or molecular processes across tissues.</p>
<p>The mixtR analysis requires gene expression profiles of two matched tissues and is based on two core functions: - <code>sig.ranksum</code> a function that maps samples to a linear ordering based on expression of a set of genes of interest - <code>stat.ranksum</code> a function that derives estimates of significance for association between gene sets of interest across tissues.</p>
<p>The mixtR package also offers ad-hoc plotting functions (<code>cohort_heatmap</code>, <code>cohort_scatterplot</code>, <code>cohort_boxplot</code>).</p>
<p>If the user wants to build a web application to explore/disseminate the results, the outputs should be saved in the <code>/data</code> folder of the R package as described below. Instructions to build the web application are described in the MIxT web application vignette.</p>
<p>First load the mixtR package in your R session.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(mixtR)</code></pre></div>
<div id="input-data-format" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data-format" class="anchor"></a>1. Input data format</h2>
<p>In this vignette, we are taking the example of gene expression profiles from microdissected epithelium and stroma from non-inflammatory invasive breast tumors from boersma et al.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">files=<span class="st"> </span><span class="kw">c</span>(<span class="st">"dat.RData"</span>, <span class="st">"moduleColors.RData"</span>)
dataDir &lt;-<span class="st"> 'https://rawgit.com/vdumeaux/mixtData/master/data/'</span>
files.dir=<span class="kw">paste0</span>(dataDir,files)

for(i in <span class="dv">1</span>:<span class="kw">length</span>(files)){
 <span class="kw">load</span>(<span class="kw">url</span>(files.dir[i]))
}</code></pre></div>
<p>Expression and clinical data from matched tissues were formatted so that they are contained in a list named <code>dat</code>.</p>
<p>Each object of the list must include the following objects: * <code>exprs</code> an expression matrix with m genes in rows and n patients in columns. Rownames of <code>exprs</code> are the same gene identifiers as the one used to define gene sets. * <code>cohorts</code> list of patients subgroups. If you have only one , the vector should still be present * <code>clinical</code> (optional) a clinical data frame with n patients in rows and clincal variables in columns. Categorical variables should be character or factors. Continuous variables should be numeric. * <code>clinical.colors</code> (optional) a clinical data frame with n patients in rows and clincal variables in columns coded by discrete or continuous color-coding. We recommend the colorspace R package that provides perceptually-based color palettes for coding categorical data (qualitative palettes) and numerical variables (sequential and diverging palettes).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(dat)</code></pre></div>
<pre><code>## List of 2
##  $ epi   :List of 4
##   ..$ exprs          : num [1:9473, 1:34] 6.12 11.65 5.54 7.55 8.31 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:9473] "3.8-1" "A2M" "A4GALT" "AACS" ...
##   .. .. ..$ : chr [1:34] "PID-1" "PID-2" "PID-4" "PID-5" ...
##   ..$ clinical       :'data.frame':  34 obs. of  9 variables:
##   .. ..$ chemo: logi [1:34] FALSE FALSE FALSE FALSE FALSE FALSE ...
##   .. ..$ er   : logi [1:34] TRUE FALSE TRUE TRUE FALSE TRUE ...
##   .. ..$ her2 : logi [1:34] FALSE FALSE FALSE FALSE FALSE FALSE ...
##   .. ..$ stage: int [1:34] 1 2 1 2 2 2 2 2 2 2 ...
##   .. ..$ event: logi [1:34] TRUE TRUE TRUE TRUE TRUE TRUE ...
##   .. ..$ MKS  : num [1:34] 5.84 6.7 5.61 5.45 7 ...
##   .. ..$ ERS  : num [1:34] 5.46 5.48 5.64 7.39 5.56 ...
##   .. ..$ LUMS : num [1:34] 6.5 6.24 6.45 8.84 6.39 ...
##   .. ..$ HER2S: num [1:34] 6.84 6.61 6.46 7.02 6.62 ...
##   ..$ clinical.colors:'data.frame':  34 obs. of  9 variables:
##   .. ..$ chemo: chr [1:34] "white" "white" "white" "white" ...
##   .. ..$ er   : chr [1:34] "green" "white" "green" "green" ...
##   .. ..$ her2 : chr [1:34] "white" "white" "white" "white" ...
##   .. ..$ stage: int [1:34] 1 2 1 2 2 2 2 2 2 2 ...
##   .. ..$ event: chr [1:34] "red" "red" "red" "red" ...
##   .. ..$ MKS  : chr [1:34] "#CEA5AC" "#9C3450" "#DED3D4" "#E2E1E1" ...
##   .. ..$ ERS  : chr [1:34] "#E1E1E2" "#E0E0E1" "#B9BDD2" "#2146A4" ...
##   .. ..$ LUMS : chr [1:34] "#CCCED9" "#E1E1E2" "#D0D1DB" "#2146A4" ...
##   .. ..$ HER2S: chr [1:34] "#B2A580" "#D8D5CE" "#E2E2E2" "#917D31" ...
##   ..$ cohorts        :List of 3
##   .. ..$ all: chr [1:34] "PID-1" "PID-2" "PID-4" "PID-5" ...
##   .. ..$ erp: chr [1:15] "PID-1" "PID-4" "PID-5" "PID-7" ...
##   .. ..$ ern: chr [1:18] "PID-2" "PID-6" "PID-12" "PID-11" ...
##  $ stroma:List of 4
##   ..$ exprs          : num [1:9473, 1:34] 5.77 11.21 5.52 7.08 7.97 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:9473] "3.8-1" "A2M" "A4GALT" "AACS" ...
##   .. .. ..$ : chr [1:34] "PID-1" "PID-2" "PID-4" "PID-5" ...
##   ..$ clinical       :'data.frame':  34 obs. of  9 variables:
##   .. ..$ chemo: logi [1:34] FALSE FALSE FALSE FALSE FALSE FALSE ...
##   .. ..$ er   : logi [1:34] TRUE FALSE TRUE TRUE FALSE TRUE ...
##   .. ..$ her2 : logi [1:34] FALSE FALSE FALSE FALSE FALSE FALSE ...
##   .. ..$ stage: int [1:34] 1 2 1 2 2 2 2 2 2 2 ...
##   .. ..$ event: logi [1:34] TRUE TRUE TRUE TRUE TRUE TRUE ...
##   .. ..$ MKS  : num [1:34] 5.84 6.7 5.61 5.45 7 ...
##   .. ..$ ERS  : num [1:34] 5.46 5.48 5.64 7.39 5.56 ...
##   .. ..$ LUMS : num [1:34] 6.5 6.24 6.45 8.84 6.39 ...
##   .. ..$ HER2S: num [1:34] 6.84 6.61 6.46 7.02 6.62 ...
##   ..$ clinical.colors:'data.frame':  34 obs. of  9 variables:
##   .. ..$ chemo: chr [1:34] "white" "white" "white" "white" ...
##   .. ..$ er   : chr [1:34] "green" "white" "green" "green" ...
##   .. ..$ her2 : chr [1:34] "white" "white" "white" "white" ...
##   .. ..$ stage: int [1:34] 1 2 1 2 2 2 2 2 2 2 ...
##   .. ..$ event: chr [1:34] "red" "red" "red" "red" ...
##   .. ..$ MKS  : chr [1:34] "#CEA5AC" "#9C3450" "#DED3D4" "#E2E1E1" ...
##   .. ..$ ERS  : chr [1:34] "#E1E1E2" "#E0E0E1" "#B9BDD2" "#2146A4" ...
##   .. ..$ LUMS : chr [1:34] "#CCCED9" "#E1E1E2" "#D0D1DB" "#2146A4" ...
##   .. ..$ HER2S: chr [1:34] "#B2A580" "#D8D5CE" "#E2E2E2" "#917D31" ...
##   ..$ cohorts        :List of 3
##   .. ..$ all: chr [1:34] "PID-1" "PID-2" "PID-4" "PID-5" ...
##   .. ..$ erp: chr [1:15] "PID-1" "PID-4" "PID-5" "PID-7" ...
##   .. ..$ ern: chr [1:18] "PID-2" "PID-6" "PID-12" "PID-11" ...</code></pre>
<p><code>moduleColors</code> is a list of length 2: each object is a named vector of gene set appartenance for each tissue. In our example, each gene is part of a module grouping strongly co-expressed genes together as computed by WGCNA R package. It could be groups of genes involved in different pathways/molecular processes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(moduleColors)</code></pre></div>
<pre><code>## List of 2
##  $ epi   : Named chr [1:9461] "darkgrey" "black" "salmon" "darkolivegreen" ...
##   ..- attr(*, "names")= chr [1:9461] "3.8-1" "A2M" "A4GALT" "AACS" ...
##  $ stroma: Named chr [1:9453] "darkgrey" "black" "cyan" "turquoise" ...
##   ..- attr(*, "names")= chr [1:9453] "3.8-1" "A2M" "A4GALT" "AACS" ...</code></pre>
<p>You can import the top node and edges for each netwrok that you can vizualize in Cytoscape or in the MIxT web application</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="kw">url</span>(<span class="st">"https://rawgit.com/vdumeaux/mixtData/master/data/TOM_net.RData"</span>))
<span class="kw">str</span>(TOM_net)</code></pre></div>
<pre><code>## List of 2
##  $ epi   :List of 2
##   ..$ edgeData:'data.frame': 111137 obs. of  6 variables:
##   .. ..$ fromNode   : Factor w/ 1729 levels "A2M","AACS","AASS",..: 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ toNode     : Factor w/ 1777 levels "ACRV1","ACSBG2",..: 5 27 104 244 263 364 393 437 451 453 ...
##   .. ..$ weight     : num [1:111137] 0.103 0.105 0.113 0.105 0.118 ...
##   .. ..$ direction  : Factor w/ 1 level "undirected": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ fromAltName: Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ toAltName  : Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   ..$ nodeData:'data.frame': 1931 obs. of  3 variables:
##   .. ..$ nodeName                : Factor w/ 1931 levels "A2M","AACS","AASS",..: 1 2 3 4 5 6 7 8 9 10 ...
##   .. ..$ altName                 : Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ nodeAttr.nodesPresent...: Factor w/ 23 levels "black","blue",..: 1 6 1 2 20 1 1 1 20 1 ...
##  $ stroma:List of 2
##   ..$ edgeData:'data.frame': 140306 obs. of  6 variables:
##   .. ..$ fromNode   : Factor w/ 1659 levels "A2M","AARSD1",..: 1 1 1 2 2 2 2 2 2 2 ...
##   .. ..$ toNode     : Factor w/ 1716 levels "ABLIM1","ACRV1",..: 549 732 836 80 172 186 252 548 1097 1378 ...
##   .. ..$ weight     : num [1:140306] 0.103 0.107 0.109 0.105 0.102 ...
##   .. ..$ direction  : Factor w/ 1 level "undirected": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ fromAltName: Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ toAltName  : Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   ..$ nodeData:'data.frame': 1865 obs. of  3 variables:
##   .. ..$ nodeName                : Factor w/ 1865 levels "A2M","AARSD1",..: 1 2 3 4 5 6 7 8 9 10 ...
##   .. ..$ altName                 : Factor w/ 1 level "NA": 1 1 1 1 1 1 1 1 1 1 ...
##   .. ..$ nodeAttr.nodesPresent...: Factor w/ 21 levels "black","blue",..: 1 19 19 12 19 19 19 19 1 12 ...</code></pre>
</div>
<div id="patient-linear-ordering-and-region-of-independence" class="section level2">
<h2 class="hasAnchor">
<a href="#patient-linear-ordering-and-region-of-independence" class="anchor"></a>2. Patient linear ordering and region of independence</h2>
<p><code><a href="../reference/sig.ranksum.html">sig.ranksum()</a></code> allows you to map ùëõ samples to a linear ordering based on expression of ùëò genes within a given gene se</p>
<p>If you know the directionality of gene expression, the gene set of interest can be split between the <code>up</code> and <code>dn</code> genes. If you do not know the directionality of gene expression, all genes in the gene set are listed in the <code>ns</code>. In this case, genes will be partitioned into two groups around myeloids using correlation as the distance metric.</p>
<p>The patient ordering is then partition in three categories (high, mid, low) following a random sampling procedure (n =10,000). By default, the region of independence (ROI) is determined by the 0.025 and 0.975 percentile point of the distribution of random patient ranks (middle.range = 0.95).Both <code>n</code> and <code>middle.range</code> can be set differently.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene.sets &lt;-<span class="st"> </span><span class="kw">lapply</span>(moduleColors, function(x.moduleColors){
  <span class="kw">split</span>(<span class="kw">names</span>(x.moduleColors), <span class="kw">factor</span>(x.moduleColors))
})
<span class="kw">str</span>(gene.sets)</code></pre></div>
<pre><code>## List of 2
##  $ epi   :List of 23
##   ..$ black         : chr [1:1438] "A2M" "AARS" "AASS" "AATF" ...
##   ..$ blue          : chr [1:710] "ABCA1" "ABCD4" "ABR" "ACAA2" ...
##   ..$ cyan          : chr [1:211] "AAMP" "ADIPOR1" "AGA" "AGL" ...
##   ..$ darkgreen     : chr [1:295] "ABCC3" "ADAM19" "ADRM1" "AHCY" ...
##   ..$ darkgrey      : chr [1:64] "3.8-1" "ALDH5A1" "ANG" "APOL1" ...
##   ..$ darkolivegreen: chr [1:433] "AACS" "ABCC5" "ABHD11" "ACD" ...
##   ..$ darkorange    : chr [1:55] "ACACA" "APEH" "ATOX1" "BNIP3" ...
##   ..$ darkred       : chr [1:165] "ADAR" "ADNP2" "AGBL5" "AGPAT1" ...
##   ..$ darkturquoise : chr [1:106] "APOBEC3A" "APOBEC3G" "APOL2" "ASPH" ...
##   ..$ green         : chr [1:902] "ABCE1" "ABCF1" "ABLIM3" "ACAT2" ...
##   ..$ greenyellow   : chr [1:142] "ADAT1" "ADORA2A" "AHNAK" "AIM2" ...
##   ..$ grey60        : chr [1:191] "ACSL5" "ADAM28" "ALDH2" "APOL3" ...
##   ..$ midnightblue  : chr [1:380] "ABAT" "ACADSB" "ACOT8" "ACOX2" ...
##   ..$ paleturquoise : chr [1:175] "ANP32B" "AP3D1" "APP" "APTX" ...
##   ..$ pink          : chr [1:280] "ABCA3" "ABCC4" "AFTPH" "ALG13" ...
##   ..$ saddlebrown   : chr [1:139] "ADK" "ADSL" "ALDH8A1" "ARFGAP1" ...
##   ..$ salmon        : chr [1:296] "A4GALT" "ABCC6" "ABHD4" "ACTL8" ...
##   ..$ steelblue     : chr [1:119] "ACLY" "ADSS" "AKAP1" "ALDH18A1" ...
##   ..$ tan           : chr [1:587] "AARSD1" "ABCC1" "ABCD3" "ABHD2" ...
##   ..$ turquoise     : chr [1:1795] "ABCA11" "ABCA12" "ABCA4" "ABCD1" ...
##   ..$ violet        : chr [1:586] "AASDHPPT" "ABCB4" "ABCC10" "ABCG1" ...
##   ..$ white         : chr [1:130] "ACE2" "AMD1" "ANXA3" "ANXA8L2" ...
##   ..$ yellow        : chr [1:262] "ABCB7" "ADCK2" "AKAP13" "AKR1A1" ...
##  $ stroma:List of 21
##   ..$ black        : chr [1:1507] "A2M" "AARS" "AASS" "ABCA6" ...
##   ..$ blue         : chr [1:706] "ABCC1" "ABCC10" "ABR" "ACADVL" ...
##   ..$ cyan         : chr [1:734] "A4GALT" "AASDHPPT" "ABCB4" "ABHD14A" ...
##   ..$ darkgreen    : chr [1:304] "AAMP" "ACOT8" "ACY1" "ADAM17" ...
##   ..$ darkgrey     : chr [1:91] "3.8-1" "ABCF1" "AK3L1" "AKAP10" ...
##   ..$ darkorange   : chr [1:263] "ADRM1" "AGRN" "AMH" "ANKRD27" ...
##   ..$ darkred      : chr [1:145] "AATF" "ADM" "AHSA1" "AKAP1" ...
##   ..$ darkturquoise: chr [1:92] "ADAR" "APOBEC3A" "APOL2" "ASS1" ...
##   ..$ green        : chr [1:498] "ABCE1" "ACAT2" "ACOX2" "ACP1" ...
##   ..$ greenyellow  : chr [1:297] "ABCA12" "ACAT1" "ACSM1" "ADORA2A" ...
##   ..$ grey60       : chr [1:180] "ABP1" "ADAM28" "AIM2" "ALG9" ...
##   ..$ lightcyan    : chr [1:185] "ABCA11" "ABCA8" "ABCB7" "ABLIM3" ...
##   ..$ midnightblue : chr [1:597] "ABCA1" "ABCC6" "ABCD3" "ABCD4" ...
##   ..$ orange       : chr [1:87] "ADAMTS1" "ADAMTS9" "ANXA1" "APOLD1" ...
##   ..$ pink         : chr [1:396] "ABCA3" "ACTN1" "ACVR1" "ACYP2" ...
##   ..$ royalblue    : chr [1:148] "ABCA5" "ABHD3" "ADCY9" "ADD1" ...
##   ..$ salmon       : chr [1:268] "ABCA7" "ACIN1" "ACOX1" "ANAPC13" ...
##   ..$ tan          : chr [1:294] "ABAT" "ABHD2" "ACADM" "ACOT9" ...
##   ..$ turquoise    : chr [1:2021] "AACS" "AARSD1" "ABCA4" "ABCC5" ...
##   ..$ white        : chr [1:66] "ANXA3" "ANXA8L2" "ARSE" "BBOX1" ...
##   ..$ yellow       : chr [1:574] "ABCC4" "ACAD8" "ACADS" "ADM2" ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## compute ranksum for each gene sets in each tissue
bresat &lt;-<span class="st"> </span><span class="kw">list</span>()

bresat$epi &lt;-<span class="st"> </span><span class="kw">lapply</span>(gene.sets$epi, function (mod){
    <span class="kw"><a href="../reference/sig.ranksum.html">sig.ranksum</a></span> (<span class="dt">x.dat =</span> dat$epi, 
                 <span class="dt">ns =</span> <span class="kw">which</span>(<span class="kw">rownames</span>(dat$epi$exprs) %in%<span class="st"> </span>mod),
                 <span class="dt">n=</span><span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">2</span>)})
  
bresat$stroma &lt;-<span class="st"> </span><span class="kw">lapply</span>(gene.sets$stroma, function (mod){
    <span class="kw"><a href="../reference/sig.ranksum.html">sig.ranksum</a></span> (<span class="dt">x.dat =</span> dat$stroma, 
                 <span class="dt">ns =</span> <span class="kw">which</span>(<span class="kw">rownames</span>(dat$stroma$exprs) %in%<span class="st"> </span>mod),
                 <span class="dt">n=</span><span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">2</span>)})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(bresat$epi)</code></pre></div>
<pre><code>##  [1] "black"          "blue"           "cyan"           "darkgreen"     
##  [5] "darkgrey"       "darkolivegreen" "darkorange"     "darkred"       
##  [9] "darkturquoise"  "green"          "greenyellow"    "grey60"        
## [13] "midnightblue"   "paleturquoise"  "pink"           "saddlebrown"   
## [17] "salmon"         "steelblue"      "tan"            "turquoise"     
## [21] "violet"         "white"          "yellow"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(bresat$epi$blue)</code></pre></div>
<pre><code>## List of 3
##  $ all:List of 11
##   ..$ pat.order : int [1:34] 1 18 16 26 3 19 5 14 29 9 ...
##   ..$ ranksum   : Named num [1:34] 31.7 20.5 23.7 10.8 21.7 ...
##   .. ..- attr(*, "names")= chr [1:34] "PID-1" "PID-2" "PID-4" "PID-5" ...
##   ..$ gene.order: int [1:710] 7569 1405 3586 1037 6219 4113 2040 4806 4450 5415 ...
##   ..$ dat       : num [1:710, 1:34] 8.41 9.18 8.8 8.73 10.12 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:710] "SLAMF8" "CD163" "HCK" "C1QA" ...
##   .. .. ..$ : chr [1:34] "PID-1" "PID-17" "PID-33" "PID-25" ...
##   ..$ up.dn     : num [1:710] 1 1 1 1 1 1 1 1 1 1 ...
##   ..$ up        : int [1:499] 12 33 52 55 62 64 73 83 84 89 ...
##   ..$ dn        : int [1:211] 90 293 304 321 354 543 623 639 645 680 ...
##   ..$ up.cor    : num [1:499] 0.736 0.4961 0.5642 0.0356 0.3445 ...
##   ..$ dn.cor    : num [1:211] 0.55 0.457 0.489 0.544 0.676 ...
##   ..$ roi       : int [1:35] 0 0 0 0 0 0 0 0 0 0 ...
##   ..$ roi.cat   : num [1:34] 3 3 3 1 3 1 1 1 3 1 ...
##  $ erp:List of 11
##   ..$ pat.order : int [1:15] 1 10 9 2 6 15 14 8 7 5 ...
##   ..$ ranksum   : Named num [1:15] 13.96 10.25 4.92 5.35 6.3 ...
##   .. ..- attr(*, "names")= chr [1:15] "PID-1" "PID-4" "PID-5" "PID-7" ...
##   ..$ gene.order: int [1:710] 719 8830 8711 5411 5133 2730 1038 390 6219 4493 ...
##   ..$ dat       : num [1:710, 1:15] 7.4 11.1 9.16 10.54 9.23 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:710] "BCL2A1" "TYROBP" "TRPV2" "NCF2" ...
##   .. .. ..$ : chr [1:15] "PID-1" "PID-17" "PID-33" "PID-4" ...
##   ..$ up.dn     : num [1:710] 1 1 1 1 1 1 1 1 1 1 ...
##   ..$ up        : int [1:500] 12 33 52 55 62 64 73 83 84 89 ...
##   ..$ dn        : int [1:210] 90 293 304 321 354 543 623 639 645 680 ...
##   ..$ up.cor    : num [1:500] 0.807 0.454 0.425 0.136 0.604 ...
##   ..$ dn.cor    : num [1:210] 0.504 0.557 0.629 0.55 0.614 ...
##   ..$ roi       : int [1:16] 0 0 0 0 0 0 0 0 0 10000 ...
##   ..$ roi.cat   : num [1:15] 3 3 1 1 1 3 1 1 3 3 ...
##  $ ern:List of 11
##   ..$ pat.order : int [1:18] 11 8 13 2 1 5 18 4 15 16 ...
##   ..$ ranksum   : Named num [1:18] 11.72 11.73 7.78 10.89 11.67 ...
##   .. ..- attr(*, "names")= chr [1:18] "PID-2" "PID-6" "PID-12" "PID-11" ...
##   ..$ gene.order: int [1:710] 1405 4806 5415 4544 1439 3976 3143 9123 5235 4113 ...
##   ..$ dat       : num [1:710, 1:18] 9.31 11.18 7.63 6.42 6.96 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:710] "CD163" "LYZ" "NCKAP1L" "LILRB2" ...
##   .. .. ..$ : chr [1:18] "PID-25" "PID-18" "PID-34" "PID-6" ...
##   ..$ up.dn     : num [1:710] 1 1 1 1 1 1 1 1 1 1 ...
##   ..$ up        : int [1:447] 12 33 52 64 73 83 84 91 105 120 ...
##   ..$ dn        : int [1:263] 55 62 89 90 117 293 304 321 354 381 ...
##   ..$ up.cor    : num [1:447] 0.6512 0.6388 0.711 0.0939 0.2074 ...
##   ..$ dn.cor    : num [1:263] 0.1166 -0.0134 0.4757 0.6533 -0.0155 ...
##   ..$ roi       : int [1:19] 0 0 0 0 0 0 0 0 7360 2640 ...
##   ..$ roi.cat   : num [1:18] 3 3 1 3 3 1 1 3 1 1 ...</code></pre>
<p>You can plot heatmap of a given gene set expressed in a given tissue and reordered by the output of sig.ranksum().</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/cohort_heatmap.html">cohort_heatmap</a></span>(<span class="dt">mixt.dat =</span> dat, <span class="dt">mixt.ranksum =</span> bresat, <span class="dt">tissue =</span> <span class="st">"epi"</span>, 
               <span class="dt">module =</span> <span class="st">"darkgrey"</span>, <span class="dt">cohort.name =</span> <span class="st">"all"</span>, <span class="dt">cl.height =</span> <span class="dv">3</span>)</code></pre></div>
<p><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-11-1.png" width="384"></p>
</div>
<div id="functional-enrichment" class="section level2">
<h2 class="hasAnchor">
<a href="#functional-enrichment" class="anchor"></a>3. Functional enrichment</h2>
<p>The MSigDB database from the Broad institute provides multiple collections of gene sets that can be used to functionally annotate the transcriptional modules.</p>
<p>MSigDB cannot be accessed without a free registration, so the first things you‚Äôll need to do is download the .gmt files using the MSigDB page and save it in a folder within yoru working directory named <code>extdata</code> for example.</p>
<p><code><a href="../reference/read.gmt.msigdb.html">read.gmt.msigdb()</a></code> will import your .gmt files in your working environment as follow</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msigdb.files &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"c1.all"</span>, <span class="st">"h.all"</span>, <span class="st">"c2.cp"</span>, <span class="st">"c2.cgp"</span>, <span class="st">"c5.all"</span>, <span class="st">"c6.all"</span>,
<span class="st">"c7.all"</span>)

msigdb &lt;-<span class="st"> </span><span class="kw">list</span>()

msigdb &lt;-<span class="st"> </span><span class="kw">lapply</span>(msigdb.files, function(p) {
  input.file &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">"../../extdata"</span>, <span class="kw">paste0</span>(p, <span class="st">".v5.2.symbols.gmt"</span>))
  ret &lt;-<span class="st">  </span><span class="kw"><a href="../reference/read.gmt.msigdb.html">read.gmt.msigdb</a></span>(input.file)
  <span class="kw">return</span>(ret)
})
<span class="kw">names</span>(msigdb) &lt;-<span class="st"> </span>msigdb.files</code></pre></div>
<p>We choose to use the hallmark (h), poistional (c1) and curated (c2) gene set collections.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sig&lt;-<span class="kw">list</span>()

sig$all.sig&lt;-<span class="kw">c</span>(msigdb$h.all, msigdb$c1.all, msigdb$c2.cp, msigdb$c2.cgp)

sig$set&lt;-<span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">"h"</span>, <span class="kw">length</span>(msigdb$h.all)),
           <span class="kw">rep</span>(<span class="st">"c1"</span>, <span class="kw">length</span>(msigdb$c1.all))
 ,          <span class="kw">rep</span>(<span class="st">"c2.cp"</span>, <span class="kw">length</span>(msigdb$c2.cp)),
           <span class="kw">rep</span>(<span class="st">"c2.cgp"</span>, <span class="kw">length</span>(msigdb$c2.cgp)))

msigdb_enrichment &lt;-<span class="st"> </span><span class="kw">list</span>()
msigdb_enrichment$epi&lt;-<span class="st"> </span><span class="kw"><a href="../reference/functional.enrichment.html">functional.enrichment</a></span>(<span class="dt">mixt.ranksum =</span> bresat,
                                              <span class="dt">tissue=</span><span class="st">"epi"</span>,
                                              <span class="dt">cohort.name =</span> <span class="st">"ern"</span>,
                                              <span class="dt">functional.groups =</span> sig,
                                              <span class="dt">p.adjust.method =</span> <span class="st">"BH"</span>,
                                              <span class="dt">mc.cores=</span><span class="dv">80</span>)

msigdb_enrichment$stroma&lt;-<span class="st"> </span><span class="kw"><a href="../reference/functional.enrichment.html">functional.enrichment</a></span>(<span class="dt">mixt.ranksum =</span> bresat,
                                                 <span class="dt">tissue=</span><span class="st">"stroma"</span>,
                                                 <span class="dt">cohort.name =</span> <span class="st">"ern"</span>,
                                                 <span class="dt">functional.groups =</span> sig,
                                                 <span class="dt">p.adjust.method =</span> <span class="st">"BH"</span>,
                                                 <span class="dt">mc.cores=</span><span class="dv">80</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(plyr)

tissue.names&lt;-<span class="kw">c</span>(<span class="st">"epi"</span>, <span class="st">"stroma"</span>)
set&lt;-<span class="kw">levels</span>(<span class="kw">factor</span>(msigdb_enrichment$epi$turquoise$results$sig.set))

results&lt;-<span class="kw">list</span>()
for (tissue in tissue.names){
results[[tissue]]&lt;-<span class="kw">lapply</span>(msigdb_enrichment[[tissue]], function(x) x$results)
}

msig.results&lt;-<span class="kw">list</span>()

msig.results&lt;-<span class="kw">sapply</span>(tissue.names, function(tissue) {
  ret &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(results[[tissue]]), function(mod) {
    res&lt;-<span class="kw">do.call</span>(rbind, <span class="kw">lapply</span>(set, function(s){
      x&lt;-results[[tissue]][[mod]][results[[tissue]][[mod]]$sig.set==s,]
      x&lt;-<span class="kw">cbind</span>(<span class="kw">data.frame</span>(<span class="dt">name=</span><span class="kw">rownames</span>(x)), x)
      top<span class="fl">.5</span>&lt;-x[<span class="kw">order</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(x$updn.pval)), <span class="dt">decreasing=</span>F),][<span class="dv">1</span>:<span class="dv">5</span>,]}))
    })
  <span class="kw">names</span>(ret)&lt;-<span class="kw">names</span>(results[[tissue]])
  <span class="kw">return</span>(ret)
  }, <span class="dt">simplify=</span>F)

epi.msig&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/plyr/topics/ldply">ldply</a></span>(msig.results$epi)
stroma.msig&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/plyr/topics/ldply">ldply</a></span>(msig.results$stroma)
knitr::<span class="kw">kable</span>(epi.msig[<span class="dv">1</span>:<span class="dv">6</span>,])</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">.id</th>
<th align="left">name</th>
<th align="left">sig.set</th>
<th align="left">sig.size</th>
<th align="left">updn.common</th>
<th align="left">updn.pval</th>
<th align="left">up.common</th>
<th align="left">up.pval</th>
<th align="left">dn.common</th>
<th align="left">dn.pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr17q25</td>
<td align="left">c1</td>
<td align="left">78</td>
<td align="left">22</td>
<td align="left">0.0277820272605066</td>
<td align="left">3</td>
<td align="left">1</td>
<td align="left">19</td>
<td align="left">0.000270500088377426</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">chr2q32</td>
<td align="left">c1</td>
<td align="left">19</td>
<td align="left">8</td>
<td align="left">0.0433681059207274</td>
<td align="left">7</td>
<td align="left">0.010963561782516</td>
<td align="left">1</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr5p15</td>
<td align="left">c1</td>
<td align="left">28</td>
<td align="left">9</td>
<td align="left">0.129887739539366</td>
<td align="left">4</td>
<td align="left">0.663248168643492</td>
<td align="left">5</td>
<td align="left">0.756451801369092</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">chr5q</td>
<td align="left">c1</td>
<td align="left">5</td>
<td align="left">3</td>
<td align="left">0.165133210864193</td>
<td align="left">3</td>
<td align="left">0.0461457585820125</td>
<td align="left">0</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr11q12</td>
<td align="left">c1</td>
<td align="left">49</td>
<td align="left">13</td>
<td align="left">0.175590118797735</td>
<td align="left">2</td>
<td align="left">1</td>
<td align="left">11</td>
<td align="left">0.0397575262064447</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">SCHUETZ_BREAST_CANCER_DUCTAL_INVASIVE_UP</td>
<td align="left">c2.cgp</td>
<td align="left">335</td>
<td align="left">183</td>
<td align="left">3.02419924062287e-61</td>
<td align="left">183</td>
<td align="left">1.07409811588174e-97</td>
<td align="left">0</td>
<td align="left">1</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(stroma.msig[<span class="dv">1</span>:<span class="dv">6</span>,])</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">.id</th>
<th align="left">name</th>
<th align="left">sig.set</th>
<th align="left">sig.size</th>
<th align="left">updn.common</th>
<th align="left">updn.pval</th>
<th align="left">up.common</th>
<th align="left">up.pval</th>
<th align="left">dn.common</th>
<th align="left">dn.pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr7p13</td>
<td align="left">c1</td>
<td align="left">18</td>
<td align="left">8</td>
<td align="left">0.0898171152669901</td>
<td align="left">3</td>
<td align="left">0.791436091301745</td>
<td align="left">5</td>
<td align="left">0.275435149832975</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">chr9p24</td>
<td align="left">c1</td>
<td align="left">17</td>
<td align="left">7</td>
<td align="left">0.173716813002666</td>
<td align="left">5</td>
<td align="left">0.176327132949814</td>
<td align="left">2</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr7q31</td>
<td align="left">c1</td>
<td align="left">39</td>
<td align="left">12</td>
<td align="left">0.205850722424551</td>
<td align="left">11</td>
<td align="left">0.0207604326983022</td>
<td align="left">1</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">chr17q25</td>
<td align="left">c1</td>
<td align="left">78</td>
<td align="left">20</td>
<td align="left">0.230751582351434</td>
<td align="left">3</td>
<td align="left">1</td>
<td align="left">17</td>
<td align="left">0.00346900028672407</td>
</tr>
<tr class="odd">
<td align="left">black</td>
<td align="left">chr14q13</td>
<td align="left">c1</td>
<td align="left">11</td>
<td align="left">5</td>
<td align="left">0.245931799726871</td>
<td align="left">5</td>
<td align="left">0.0392594642649441</td>
<td align="left">0</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">black</td>
<td align="left">SWEET_LUNG_CANCER_KRAS_DN</td>
<td align="left">c2.cgp</td>
<td align="left">316</td>
<td align="left">133</td>
<td align="left">6.75526042336947e-26</td>
<td align="left">126</td>
<td align="left">6.92217507816109e-43</td>
<td align="left">7</td>
<td align="left">1</td>
</tr>
</tbody>
</table>
<p>We also annotate gene sets/modules based on gene ontology (GO) enrichment, for example using the R topGO package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(topGO)
<span class="kw">library</span>(org.Hs.eg.db)
<span class="kw">library</span>(plyr)
bckg.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(dat[[<span class="dv">1</span>]]$exprs) 
xx &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/annFUN">annFUN.org</a></span>(<span class="st">"BP"</span>, <span class="dt">mapping =</span> <span class="st">"org.Hs.eg.db"</span>, <span class="dt">ID =</span> <span class="st">"symbol"</span>)
<span class="kw">head</span>(xx)
xx &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">unlist</span>(xx))
xx&lt;-xx[xx %in%<span class="st"> </span>bckg.genes]

goterms &lt;-<span class="st"> </span><span class="ot">NULL</span>

goterms &lt;-<span class="st"> </span><span class="kw">lapply</span>(tissue.names, function (tissue){
  
  ret &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(gene.sets[[tissue]]), function(mod) {
  gene.list &lt;-<span class="st"> </span>gene.sets[[tissue]][[mod]]
  gene.list<span class="fl">.2</span> &lt;-<span class="st"> </span><span class="kw">ifelse</span>(xx %in%<span class="st"> </span>gene.list, <span class="dv">1</span>, <span class="dv">0</span>)
  gene.list<span class="fl">.2</span> &lt;-<span class="st"> </span><span class="kw">factor</span>(gene.list<span class="fl">.2</span>)
  <span class="kw">names</span>(gene.list<span class="fl">.2</span>) &lt;-<span class="st"> </span>xx
    
  GO.genes &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">"topGOdata"</span>,
                    <span class="dt">ontology =</span> <span class="st">"BP"</span>,
                    <span class="dt">allGenes =</span> gene.list<span class="fl">.2</span>,
                    <span class="dt">nodeSize =</span> <span class="dv">5</span>,
                    <span class="dt">annot =</span> annFUN.org, 
                    <span class="dt">mapping =</span> <span class="st">"org.Hs.eg.db"</span>,
                    <span class="dt">ID =</span> <span class="st">"symbol"</span>)
    
  resultFisher &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/getSigGroups">runTest</a></span>(GO.genes, <span class="dt">algorithm =</span> <span class="st">"classic"</span>, 
                            <span class="dt">statistic =</span> <span class="st">"fisher"</span>)
    
  weight01Fisher &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/getSigGroups">runTest</a></span>(GO.genes, <span class="dt">statistic =</span> <span class="st">"fisher"</span>)
    
  allRes &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/diagnosticMethods">GenTable</a></span>(GO.genes, <span class="dt">classicFisher =</span> resultFisher, 
                      <span class="dt">weight01Fisher=</span> weight01Fisher, 
                      <span class="dt">orderBy =</span> <span class="st">"weight01Fisher"</span>, 
                      <span class="dt">topNodes=</span><span class="kw">length</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/topGOdata-class">usedGO</a></span>(GO.genes)), <span class="dt">numChar =</span> <span class="dv">1000</span>) 
  allRes &lt;-<span class="st"> </span><span class="kw">subset</span>(allRes, classicFisher !=<span class="st"> "1.00000"</span>)
    
  gt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/topGO/topics/topGOdata-class">genesInTerm</a></span>(GO.genes, allRes$GO.ID)
  mod.genes &lt;-<span class="st"> </span>gene.sets[[tissue]][[mod]]
  common&lt;-<span class="kw">lapply</span>(gt, function(sig) <span class="kw">intersect</span>(mod.genes, sig))
    
  results &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">GO.table =</span> allRes,
              <span class="dt">common =</span> common)

  <span class="kw">return</span>(results)
})
  <span class="kw">names</span>(ret) &lt;-<span class="st"> </span><span class="kw">names</span>(gene.sets[[tissue]])
  <span class="kw">return</span>(ret)
})
<span class="kw">names</span>(goterms) &lt;-<span class="st"> </span>tissue.names

goterms.table &lt;-<span class="st"> </span><span class="kw">list</span>()
for (tissue in <span class="kw">c</span>(<span class="st">"epi"</span>, <span class="st">"stroma"</span>)){
  goterms.table[[tissue]] &lt;-<span class="st"> </span><span class="kw">lapply</span>(goterms[[tissue]], <span class="st">"[["</span>, <span class="st">"GO.table"</span>)
  goterms.table[[tissue]] &lt;-<span class="st"> </span><span class="kw">lapply</span>(goterms.table[[tissue]], function(x){
    dplyr::<span class="kw">slice</span>(x, <span class="dv">1</span>:<span class="dv">20</span>)})
}

epi.goterms&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/plyr/topics/ldply">ldply</a></span>(goterms.table$epi)
stroma.goterms&lt;-<span class="kw"><a href="http://www.rdocumentation.org/packages/plyr/topics/ldply">ldply</a></span>(goterms.table$stroma)</code></pre></div>
</div>
<div id="association-between-gene-sets-and-clinical-variables" class="section level1">
<h1 class="hasAnchor">
<a href="#association-between-gene-sets-and-clinical-variables" class="anchor"></a>4. Association between gene sets and clinical variables</h1>
<p>Using ranksums to capture module expression, we can ask how gene sets/modules in each tissue are differentially expressed according to patient‚Äôs clinicopathological variables. The type of the clinicopathological attribute (categorical or continuous) determines the underlying statistical test. Pearson correlation (Student asymptotic p-value) is used to test association between between a given module and continuous patient attributes (eg. age). Analysis of Variance (ANOVA) is used to test association between a given module and categorical patient attributes (eg. ER status).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">epi.clinical.assoc &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(dat$epi$cohorts), function(cohort.name) {
  <span class="kw"><a href="../reference/clinical.ranksum.html">clinical.ranksum</a></span>(<span class="dt">mixt.dat =</span> dat, 
                   <span class="dt">mixt.ranksum =</span> bresat, 
                   <span class="dt">tissue=</span><span class="st">"epi"</span>, <span class="dt">cohort =</span> cohort.name,
                   <span class="dt">nRuns =</span> <span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">80</span>)
})
<span class="kw">names</span>(epi.clinical.assoc) &lt;-<span class="st"> </span><span class="kw">names</span>(dat$epi$cohorts)
knitr::<span class="kw">kable</span>(epi.clinical.assoc$ern[, <span class="dv">1</span>:<span class="dv">6</span>])

stroma.clinical.assoc &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(dat$stroma$cohorts), function(cohort.name) {
  <span class="kw"><a href="../reference/clinical.ranksum.html">clinical.ranksum</a></span>(<span class="dt">mixt.dat =</span> dat, 
                   <span class="dt">mixt.ranksum =</span> bresat,
                   <span class="dt">tissue=</span><span class="st">"stroma"</span>, <span class="dt">cohort=</span>cohort.name,
                   <span class="dt">nRuns =</span> <span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">80</span>)
})
<span class="kw">names</span>(stroma.clinical.assoc) &lt;-<span class="st"> </span><span class="kw">names</span>(dat$stroma$cohorts)
knitr::<span class="kw">kable</span>(stroma.clinical.assoc$all[, <span class="dv">1</span>:<span class="dv">6</span>])

## Adjust for mutliple testing
emp.p &lt;-<span class="st"> </span><span class="kw">list</span>()

emp.p &lt;-<span class="st"> </span><span class="ot">NULL</span>
for (cohort.name in <span class="kw">names</span>(dat$epi$cohorts)){
  emp.p[[cohort.name]] &lt;-<span class="st"> </span><span class="kw">cbind</span>(epi.clinical.assoc[[cohort.name]], stroma.clinical.assoc[[cohort.name]])
}

### Families of tests
group1 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"her2"</span>, <span class="st">"HER2S"</span>)
group2 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"er"</span>, <span class="st">"LUMS"</span>, <span class="st">"ERS"</span>)
singles &lt;-<span class="st"> </span><span class="kw">rownames</span>(emp.p$all)[!<span class="kw">rownames</span>(emp.p$all) %in%<span class="st"> </span><span class="kw">c</span>(group1, group2) ]
gp &lt;-<span class="st"> </span><span class="kw">list</span>(group1, group2)

mod_clinical_fdr &lt;-<span class="st"> </span><span class="ot">NULL</span>
mod_clinical_fdr &lt;-<span class="st"> </span><span class="kw">lapply</span>(emp.p, function(p.dat){
  ret &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, <span class="kw">lapply</span>(gp, function(gpx) {
      <span class="kw">matrix</span>(<span class="kw">p.adjust</span>(p.dat[<span class="kw">rownames</span>(p.dat) %in%<span class="st"> </span>gpx, ], <span class="dt">method=</span><span class="st">"BH"</span>), 
             <span class="dt">nrow=</span><span class="kw">length</span>(<span class="kw">which</span>(<span class="kw">rownames</span>(p.dat) %in%<span class="st"> </span>gpx)), 
             <span class="dt">ncol =</span> <span class="kw">ncol</span>(p.dat),
             <span class="dt">dimnames =</span> <span class="kw">list</span>(<span class="kw">rownames</span>(p.dat)[<span class="kw">rownames</span>(p.dat) %in%<span class="st"> </span>gpx], <span class="kw">colnames</span>(p.dat))
             )
    }))
  ret &lt;-<span class="st"> </span><span class="kw">rbind</span>(ret, <span class="kw">t</span>(<span class="kw">apply</span>(p.dat[<span class="kw">rownames</span>(p.dat) %in%<span class="st"> </span>singles, ], <span class="dv">1</span>, p.adjust, <span class="dt">method=</span><span class="st">"BH"</span>)))
  <span class="kw">return</span>(ret)
})


mod_clinical_fdr &lt;-<span class="st"> </span><span class="kw">lapply</span>(mod_clinical_fdr, function(val){
  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">epi=</span>val[, <span class="dv">1</span>:<span class="dv">23</span>], <span class="dt">stroma=</span>val[,<span class="dv">24</span>:<span class="dv">44</span>]))
})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(mod_clinical_fdr)</code></pre></div>
<pre><code>## List of 3
##  $ all:List of 2
##   ..$ epi   : num [1:9, 1:23] 0.615 0.429 0.838 0.693 0.282 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:9] "her2" "HER2S" "er" "ERS" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   ..$ stroma: num [1:9, 1:21] 0.862 0.862 0.353 0.625 0.878 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:9] "her2" "HER2S" "er" "ERS" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##  $ erp:List of 2
##   ..$ epi   : num [1:7, 1:23] 0.7781 0.2617 0.1265 0.9873 0.0295 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:7] "HER2S" "ERS" "LUMS" "chemo" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   ..$ stroma: num [1:7, 1:21] 0.931 1 0.676 0.987 0.364 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:7] "HER2S" "ERS" "LUMS" "chemo" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##  $ ern:List of 2
##   ..$ epi   : num [1:8, 1:23] 0.703 0.481 0.769 0.717 0.844 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:8] "her2" "HER2S" "ERS" "LUMS" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   ..$ stroma: num [1:8, 1:21] 0.893 0.892 0.945 0.945 0.844 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:8] "her2" "HER2S" "ERS" "LUMS" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(mod_clinical_fdr$all)</code></pre></div>
<pre><code>## $epi
##            black      blue       cyan darkgreen   darkgrey darkolivegreen
## her2  0.61549381 0.5460737 0.42892710 0.9218922 0.54685469     0.42892710
## HER2S 0.42892710 0.5780418 0.20902090 0.9218922 0.71847185     0.24290429
## er    0.83753767 0.8452259 0.26440358 0.9872987 0.23348604     0.96122112
## ERS   0.69319873 0.0345564 0.90565420 0.2339886 0.05313531     0.11119573
## LUMS  0.28150421 0.0120012 0.80515942 0.1363336 0.02895773     0.01320132
## chemo 0.93775693 0.9377569 0.93775693 0.9377569 0.93775693     0.93775693
## stage 0.54334957 0.3491016 0.72431243 0.5433496 0.76817411     0.37403740
## event 0.90447902 0.9044790 0.90447902 0.9373937 0.90447902     0.90447902
## MKS   0.02510839 0.6903662 0.01210121 0.3359003 0.03798275     0.46707528
##       darkorange    darkred darkturquoise       green greenyellow
## her2  0.42892710 0.87456881     0.6383388 0.458085809  0.96300714
## HER2S 0.20902090 0.96300714     0.8622907 0.428927103  0.92895605
## er    0.54383369 0.68924092     0.6892409 0.006948063  0.12687935
## ERS   0.28898232 0.39072688     0.6941587 0.006600660  0.08718372
## LUMS  0.06820682 0.34062880     0.5927393 0.034556397  0.18879983
## chemo 0.85808581 0.93775693     0.9377569 0.937756934  0.93775693
## stage 0.34910158 0.34910158     0.7484873 0.592528818  0.54334957
## event 0.90447902 0.90447902     0.9044790 0.906061338  0.90447902
## MKS   0.62832950 0.01963273     0.9883818 0.000000000  0.27384277
##           grey60 midnightblue paleturquoise       pink saddlebrown
## her2  0.42892710   0.57804180    0.92189219 0.86933138   0.9218922
## HER2S 0.42892710   0.42892710    0.86933138 0.92189219   0.9280709
## er    0.79780808   0.05313531    0.35285010 0.02053539   0.9872987
## ERS   0.15819209   0.00000000    0.69560802 0.00000000   0.6022385
## LUMS  0.07260726   0.00000000    0.66405391 0.00000000   0.4515166
## chemo 0.93775693   0.94767425    0.93775693 0.93775693   0.9377569
## stage 0.72431243   0.54334957    0.77865222 0.54334957   0.9789444
## event 0.90447902   0.90447902    0.90447902 0.90447902   0.9060613
## MKS   0.81801864   0.01466813    0.01880188 0.02172717   0.9883818
##          salmon  steelblue        tan turquoise     violet       white
## her2  0.9218922 0.42892710 0.42892710 0.8693314 0.86229067 0.987013644
## HER2S 0.8622907 0.20902090 0.42892710 0.9289561 0.59785289 0.921892189
## er    0.6892409 0.83675153 0.00120012 0.8367515 0.90143014 0.002030972
## ERS   0.8972460 0.90612996 0.00000000 0.3482475 0.05313531 0.003771806
## LUMS  0.8367515 0.47431567 0.00000000 0.2206221 0.01265127 0.000000000
## chemo 0.9377569 0.93775693 0.93775693 0.9377569 0.99369937 0.993699370
## stage 0.7875714 0.54334957 0.72431243 0.7484873 0.54334957 0.787571440
## event 0.9044790 0.90447902 0.90447902 0.9060613 0.90606134 0.904479019
## MKS   0.9948995 0.06440644 0.00000000 0.8180186 0.62832950 0.011944052
##           yellow
## her2  0.28016135
## HER2S 0.20902090
## er    0.34062880
## ERS   0.60223848
## LUMS  0.83753767
## chemo 0.93775693
## stage 0.54334957
## event 0.90447902
## MKS   0.05406255
## 
## $stroma
##            black       blue       cyan   darkgreen   darkgrey darkorange
## her2  0.86229067 0.96478354 0.87456881 0.869331378 0.86933138 0.96300714
## HER2S 0.86229067 0.87097281 0.87456881 0.963007144 0.96300714 0.86933138
## er    0.35285010 0.56405641 0.19760726 0.805159415 0.06846266 0.08379968
## ERS   0.62486249 0.05313531 0.96122112 0.002030972 0.20309723 0.92803151
## LUMS  0.87788779 0.03734088 0.96122112 0.012651265 0.12055545 0.86011678
## chemo 0.93775693 0.93775693 0.93775693 0.937756934 0.93775693 0.93775693
## stage 0.76751008 0.72431243 0.56585659 0.543349573 0.54334957 0.69361936
## event 0.90447902 0.90447902 0.90447902 0.904479019 0.90447902 0.90606134
## MKS   0.01963273 0.62832950 0.01628163 0.818018644 0.04466447 0.02172717
##           darkred darkturquoise       green greenyellow     grey60
## her2  0.963007144     0.9647835 0.862290674  0.61166117 0.99549955
## HER2S 0.921892189     0.8693314 0.869331378  0.42892710 0.96300714
## er    0.164908294     0.2889823 0.010058149  0.04730473 0.23398862
## ERS   0.987298730     0.6345060 0.004125413  0.02593116 0.02895773
## LUMS  0.837537667     0.6635400 0.008580858  0.15181518 0.06600660
## chemo 0.937756934     0.9377569 0.937756934  0.99369937 0.93775693
## stage 0.543349573     0.5433496 0.981898190  0.72431243 0.76751008
## event 0.906061338     0.9044790 0.904479019  0.90447902 0.90447902
## MKS   0.009534287     0.4876901 0.000000000  0.07270292 0.22671067
##        lightcyan midnightblue    orange       pink  royalblue    salmon
## her2  0.42892710  0.862290674 0.9870136 0.86516478 0.59785289 0.5978529
## HER2S 0.57804180  0.862290674 0.9218922 0.87097281 0.81511681 0.9289561
## er    0.10402640  0.801085716 0.1518152 0.32396753 0.14403583 0.6892409
## ERS   0.11119573  0.027768294 0.5300484 0.09806695 0.00120012 0.4383474
## LUMS  0.08718372  0.004659289 0.5658768 0.03455640 0.01675552 0.9872987
## chemo 0.93775693  0.978722262 0.9377569 0.93775693 0.93775693 0.9377569
## stage 0.54334957  0.543349573 0.5433496 0.76751008 0.76567657 0.8099953
## event 0.90447902  0.904479019 0.9044790 0.90447902 0.90447902 0.9373937
## MKS   0.03618140  0.994899490 0.1241291 0.99489949 0.00000000 0.9647119
##               tan turquoise      white    yellow
## her2  0.862290674 0.9218922 0.92189219 0.5978529
## HER2S 0.862290674 0.9218922 0.92189219 0.8622907
## er    0.004125413 0.3408341 0.08379968 0.9280315
## ERS   0.000000000 0.3462808 0.17183009 0.7356278
## LUMS  0.000000000 0.9540858 0.16490829 0.6931987
## chemo 0.937756934 0.9377569 0.93775693 0.9787223
## stage 0.543349573 0.5433496 0.77865222 0.7243124
## event 0.937393739 0.9044790 0.90447902 0.9044790
## MKS   0.005280528 0.6903662 0.02043061 0.6283295</code></pre>
<div id="investigating-gene-overlap-between-gene-sets-across-tissues" class="section level2">
<h2 class="hasAnchor">
<a href="#investigating-gene-overlap-between-gene-sets-across-tissues" class="anchor"></a>5. Investigating gene overlap between gene sets across tissues</h2>
<p>We first exclude genes that are not present in both tissue and check significance of overlap between modules across tissue using a Fisher exact test.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">overlapFun &lt;-<span class="st"> </span>function (labels1, labels2)
{
    labels1 &lt;-<span class="st">  </span><span class="kw">as.vector</span>(labels1)
    labels2 &lt;-<span class="st">  </span><span class="kw">as.vector</span>(labels2)
    
    levels1 &lt;-<span class="st">  </span><span class="kw">sort</span>(<span class="kw">unique</span>(labels1))
    levels2 &lt;-<span class="st">  </span><span class="kw">sort</span>(<span class="kw">unique</span>(labels2))

    n1 &lt;-<span class="st">  </span><span class="kw">length</span>(levels1)
    n2 &lt;-<span class="st">  </span><span class="kw">length</span>(levels2)
    
    countMat &lt;-<span class="st">   </span><span class="kw">matrix</span>(<span class="dv">0</span>, n1, n2)
    pMat &lt;-<span class="st">  </span><span class="kw">matrix</span>(<span class="dv">0</span>, n1, n2)
    
    for (m1 in <span class="dv">1</span>:n1) for (m2 in <span class="dv">1</span>:n2) {
        m1Members =<span class="st"> </span>(labels1 ==<span class="st"> </span>levels1[m1])
        m2Members =<span class="st"> </span>(labels2 ==<span class="st"> </span>levels2[m2])
        
        t &lt;-<span class="st"> </span><span class="kw">table</span>(m1Members, m2Members)
        <span class="co"># make sure all levels are included</span>
        tab &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">2</span>)
        tab[<span class="kw">match</span>(<span class="kw">rownames</span>(t), <span class="kw">c</span>(<span class="ot">FALSE</span>, <span class="ot">TRUE</span>)), <span class="kw">match</span>(<span class="kw">colnames</span>(t), <span class="kw">c</span>(<span class="ot">FALSE</span>, <span class="ot">TRUE</span>))] &lt;-<span class="st"> </span>t

        pMat[m1, m2] =<span class="st"> </span><span class="kw">fisher.test</span>(tab, <span class="dt">alternative =</span> <span class="st">"greater"</span>)$p.value
        countMat[m1, m2] =<span class="st"> </span><span class="kw">sum</span>(labels1 ==<span class="st"> </span>levels1[m1] &amp;<span class="st"> </span>labels2 ==
<span class="st">            </span>levels2[m2])
    }
    <span class="kw">dimnames</span>(pMat) =<span class="st"> </span><span class="kw">list</span>(levels1, levels2)
    <span class="kw">dimnames</span>(countMat) =<span class="st"> </span><span class="kw">list</span>(levels1, levels2)
    pMat[<span class="kw">is.na</span>(pMat)] =<span class="st"> </span><span class="dv">1</span>
    <span class="kw">list</span>(<span class="dt">countTable =</span> countMat, <span class="dt">pTable =</span> pMat)
}

<span class="kw">str</span>(moduleColors)</code></pre></div>
<pre><code>## List of 2
##  $ epi   : Named chr [1:9461] "darkgrey" "black" "salmon" "darkolivegreen" ...
##   ..- attr(*, "names")= chr [1:9461] "3.8-1" "A2M" "A4GALT" "AACS" ...
##  $ stroma: Named chr [1:9453] "darkgrey" "black" "cyan" "turquoise" ...
##   ..- attr(*, "names")= chr [1:9453] "3.8-1" "A2M" "A4GALT" "AACS" ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">modColors &lt;-<span class="st"> </span><span class="kw">list</span>()
modColors$epi &lt;-<span class="st"> </span>moduleColors$epi[<span class="kw">names</span>(moduleColors$epi) %in%<span class="st"> </span><span class="kw">names</span>(moduleColors$stroma)]
modColors$stroma &lt;-<span class="st"> </span>moduleColors$stroma[<span class="kw">names</span>(moduleColors$stroma) %in%<span class="st"> </span><span class="kw">names</span>(moduleColors$epi)]
<span class="kw">str</span>(modColors)</code></pre></div>
<pre><code>## List of 2
##  $ epi   : Named chr [1:9441] "darkgrey" "black" "salmon" "darkolivegreen" ...
##   ..- attr(*, "names")= chr [1:9441] "3.8-1" "A2M" "A4GALT" "AACS" ...
##  $ stroma: Named chr [1:9441] "darkgrey" "black" "cyan" "turquoise" ...
##   ..- attr(*, "names")= chr [1:9441] "3.8-1" "A2M" "A4GALT" "AACS" ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene.set.overlap &lt;-<span class="st">  </span><span class="kw">overlapFun</span>(modColors[[<span class="dv">1</span>]], modColors[[<span class="dv">2</span>]])
<span class="kw">str</span>(gene.set.overlap)</code></pre></div>
<pre><code>## List of 2
##  $ countTable: num [1:23, 1:21] 648 47 16 32 6 108 7 13 7 128 ...
##   ..- attr(*, "dimnames")=List of 2
##   .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##  $ pTable    : num [1:23, 1:21] 2.62e-188 1.00 1.00 9.96e-01 9.55e-01 ...
##   ..- attr(*, "dimnames")=List of 2
##   .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...</code></pre>
<p>P-values are adjusted for multiple testing using the false discovery rate of (Benjamini and Hochberg, 1995).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene.set.overlap$pTable &lt;-<span class="st"> </span><span class="kw">matrix</span>(
  <span class="kw">p.adjust</span>(gene.set.overlap$pTable, <span class="dt">method =</span> <span class="st">"BH"</span>),
  <span class="dt">nrow =</span> <span class="kw">nrow</span>(gene.set.overlap$pTable),
  <span class="dt">ncol =</span> <span class="kw">ncol</span>(gene.set.overlap$pTable), 
  <span class="dt">dimnames =</span> <span class="kw">list</span>(<span class="kw">rownames</span>(gene.set.overlap$pTable),
                  <span class="kw">colnames</span>(gene.set.overlap$pTable))
  )
                                  
knitr::<span class="kw">kable</span>(gene.set.overlap$pTable[<span class="dv">1</span>:<span class="dv">10</span>, <span class="dv">1</span>:<span class="dv">8</span>])</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">black</th>
<th align="right">blue</th>
<th align="right">cyan</th>
<th align="right">darkgreen</th>
<th align="right">darkgrey</th>
<th align="right">darkorange</th>
<th align="right">darkred</th>
<th align="right">darkturquoise</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>black</td>
<td align="right">0.0e+00</td>
<td align="right">1</td>
<td align="right">1.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">0.1960916</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>blue</td>
<td align="right">1.0e+00</td>
<td align="right">0</td>
<td align="right">1.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>cyan</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">0.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">0.0213872</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>darkgreen</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000000</td>
<td align="right">1</td>
<td align="right">0.0000000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>darkgrey</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">1.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>darkolivegreen</td>
<td align="right">5.1e-06</td>
<td align="right">1</td>
<td align="right">1.00e+00</td>
<td align="right">0.0292844</td>
<td align="right">1</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>darkorange</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">1.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">0.0000000</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>darkred</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">1.39e-05</td>
<td align="right">0.3152854</td>
<td align="right">1</td>
<td align="right">1.0000000</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>darkturquoise</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">1.00e+00</td>
<td align="right">1.0000000</td>
<td align="right">0</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>green</td>
<td align="right">1.0e+00</td>
<td align="right">1</td>
<td align="right">4.47e-05</td>
<td align="right">1.0000000</td>
<td align="right">1</td>
<td align="right">0.0032507</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div id="matched-interactions-across-tissues" class="section level2">
<h2 class="hasAnchor">
<a href="#matched-interactions-across-tissues" class="anchor"></a>6. Matched Interactions Across Tissues</h2>
<p>Interactions between gene sets/modules across tissues are identified using a random permutation approach based on the Pearson correlation between ranksums of gene expression in gene sets/modules across tissues. This is run independently within each patient cohort.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">perm_cor_p &lt;-<span class="st"> </span><span class="ot">NULL</span>

perm_cor_p$epi2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/stat.ranksum.html">stat.ranksum</a></span> (<span class="dt">mixt.ranksum =</span> bresat, 
                                       <span class="dt">tissue1=</span><span class="st">"epi"</span>, <span class="dt">tissue2=</span><span class="st">"epi"</span>,
                                       <span class="dt">nRuns =</span> <span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">80</span>)
perm_cor_p$epi_stroma &lt;-<span class="st"> </span><span class="kw"><a href="../reference/stat.ranksum.html">stat.ranksum</a></span> (<span class="dt">mixt.ranksum =</span> bresat, 
                                       <span class="dt">tissue1=</span><span class="st">"epi"</span>, <span class="dt">tissue2=</span><span class="st">"stroma"</span>,
                                       <span class="dt">nRuns =</span> <span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">80</span>)
perm_cor_p$stroma2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/stat.ranksum.html">stat.ranksum</a></span> (<span class="dt">mixt.ranksum =</span> bresat, 
                                       <span class="dt">tissue1=</span><span class="st">"stroma"</span>, <span class="dt">tissue2=</span><span class="st">"stroma"</span>,
                                       <span class="dt">nRuns =</span> <span class="dv">10000</span>, <span class="dt">mc.cores =</span> <span class="dv">80</span>)</code></pre></div>
<p>Gene sets from tissue 1 (<code>epi</code>) are in row and gene sets from tissue 2 (<code>stroma</code>) are in column.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(perm_cor_p)</code></pre></div>
<pre><code>## List of 3
##  $ epi2      :List of 3
##   ..$ all: num [1:23, 1:23] 0 0.02 0.0008 0.0002 0.2239 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   ..$ erp: num [1:23, 1:23] 0 0.1102 0.032 0.0058 0.3088 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   ..$ ern: num [1:23, 1:23] 0 0.0827 0.0076 0.0075 0.5438 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##  $ epi_stroma:List of 3
##   ..$ all: num [1:23, 1:21] 0 0.4521 0.0857 0.0416 0.02 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   ..$ erp: num [1:23, 1:21] 0.0077 0.9455 0.3603 0.253 0.019 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   ..$ ern: num [1:23, 1:21] 0.0002 0.2534 0.2767 0.1016 0.5848 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:23] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##  $ stroma2   :List of 3
##   ..$ all: num [1:21, 1:21] 0 0.2514 0.0588 0.0181 0.0011 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   ..$ erp: num [1:21, 1:21] 0 0.382 0.369 0.0442 0.0067 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   ..$ ern: num [1:21, 1:21] 0 0.594 0.21 0.195 0.192 ...
##   .. ..- attr(*, "dimnames")=List of 2
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...
##   .. .. ..$ : chr [1:21] "black" "blue" "cyan" "darkgreen" ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(perm_cor_p$epi_stroma$all)</code></pre></div>
<pre><code>##                     black       blue       cyan  darkgreen   darkgrey
## black          0.00000000 0.61066107 0.07250725 0.00740074 0.03120312
## blue           0.45214521 0.00000000 0.00380038 0.04200420 0.02450245
## cyan           0.08570857 0.84658466 0.00000000 0.03580358 0.35693569
## darkgreen      0.04160416 0.31323132 0.00000000 0.00020002 0.82698270
## darkgrey       0.02000200 0.00190019 0.77977798 0.89318932 0.00000000
## darkolivegreen 0.01990199 0.03710371 0.02740274 0.01740174 0.51035104
##                darkorange    darkred darkturquoise      green greenyellow
## black          0.00020002 0.06140614     0.0530053 0.01640164  0.04320432
## blue           0.31043104 0.04090409     0.1871187 0.46064606  0.27942794
## cyan           0.00210021 0.01160116     0.5711571 0.00400040  0.94409441
## darkgreen      0.00160016 0.01200120     0.7443744 0.10281028  0.65956596
## darkgrey       0.03850385 0.17631763     0.0000000 0.02790279  0.04040404
## darkolivegreen 0.02410241 0.28422842     0.9757976 0.24562456  0.68746875
##                    grey60  lightcyan midnightblue     orange       pink
## black          0.64296430 0.12391239   0.01830183 0.07650765 0.00750075
## blue           0.02430243 0.65376538   0.00000000 0.69626963 0.05300530
## cyan           0.53815382 0.12251225   0.03610361 0.62836284 0.88798880
## darkgreen      0.11991199 0.54825483   0.00230023 0.97639764 0.22452245
## darkgrey       0.00030003 0.03930393   0.22492249 0.02370237 0.81898190
## darkolivegreen 0.14241424 0.73647365   0.00020002 0.67786779 0.03200320
##                 royalblue     salmon        tan  turquoise     white
## black          0.12101210 0.08450845 0.28612861 0.60286029 0.3553355
## blue           0.46754675 0.28052805 0.00780078 0.41334133 0.1863186
## cyan           0.20122012 0.61936194 0.60966097 0.05600560 0.5490549
## darkgreen      0.86038604 0.96659666 0.12671267 0.08910891 0.9650965
## darkgrey       0.06920692 0.02630263 0.05380538 0.67066707 0.9772977
## darkolivegreen 0.82958296 0.90119012 0.02510251 0.43464346 0.3286329
##                    yellow
## black          0.85438544
## blue           0.13771377
## cyan           0.01220122
## darkgreen      0.09020902
## darkgrey       0.33873387
## darkolivegreen 0.57765777</code></pre>
<p>You can then examine these associations with different plots</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/cohort_scatterplot.html">cohort_scatterplot</a></span>(<span class="dt">mixt.ranksum =</span> bresat, 
                   <span class="dt">x.tissue =</span> <span class="st">"epi"</span>, <span class="dt">x.module =</span> <span class="st">"darkgrey"</span>, 
                   <span class="dt">y.tissue =</span> <span class="st">"stroma"</span>, <span class="dt">y.module =</span> <span class="st">"grey60"</span>, 
                   <span class="dt">cohort.name =</span> <span class="st">"ern"</span>)

<span class="kw"><a href="../reference/cohort_scatterplot.html">cohort_scatterplot</a></span>(<span class="dt">mixt.ranksum =</span> bresat, 
                   <span class="dt">x.tissue =</span> <span class="st">"epi"</span>, <span class="dt">x.module =</span> <span class="st">"darkgrey"</span>, 
                   <span class="dt">y.tissue =</span> <span class="st">"epi"</span>, <span class="dt">y.module =</span> <span class="st">"grey60"</span>, 
                   <span class="dt">cohort.name =</span> <span class="st">"ern"</span>)

<span class="kw"><a href="../reference/cohort_scatterplot.html">cohort_scatterplot</a></span>(<span class="dt">mixt.ranksum =</span> bresat, 
                   <span class="dt">x.tissue =</span> <span class="st">"stroma"</span>, <span class="dt">x.module =</span> <span class="st">"darkgrey"</span>, 
                   <span class="dt">y.tissue =</span> <span class="st">"stroma"</span>, <span class="dt">y.module =</span> <span class="st">"grey60"</span>, 
                   <span class="dt">cohort.name =</span> <span class="st">"ern"</span>)</code></pre></div>
<p><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-32-1.png" width="384"><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-32-2.png" width="384"><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-32-3.png" width="384"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/cohort_heatmap.html">cohort_heatmap</a></span>(<span class="dt">mixt.dat =</span> dat, <span class="dt">mixt.ranksum =</span> bresat, <span class="dt">tissue =</span> <span class="st">"stroma"</span>, 
               <span class="dt">module =</span> <span class="st">"grey60"</span>, <span class="dt">cohort.name =</span> <span class="st">"ern"</span>, <span class="dt">cl.height =</span> <span class="dv">3</span>)</code></pre></div>
<p><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-33-1.png" width="384"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/cohort_heatmap.html">cohort_heatmap</a></span>(<span class="dt">mixt.dat =</span> dat, <span class="dt">mixt.ranksum =</span> bresat, <span class="dt">tissue =</span> <span class="st">"epi"</span>, 
               <span class="dt">module =</span> <span class="st">"darkgrey"</span>, <span class="dt">cohort.name =</span> <span class="st">"ern"</span>, <span class="dt">cl.height =</span> <span class="dv">3</span>, 
               <span class="dt">orderByTissue =</span> <span class="st">"stroma"</span>, <span class="dt">orderByModule =</span> <span class="st">"grey60"</span>)</code></pre></div>
<p><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-34-1.png" width="384"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/cohort_boxplot.html">cohort_boxplot</a></span>(<span class="dt">mixt.ranksum =</span> bresat, <span class="dt">tissue =</span> <span class="st">"epi"</span>, <span class="dt">module =</span> <span class="st">"darkgrey"</span>, 
               <span class="dt">cohort.name =</span> <span class="st">"ern"</span>,
               <span class="dt">orderByTissue =</span> <span class="st">"stroma"</span>, <span class="dt">orderByModule =</span> <span class="st">"grey60"</span>)</code></pre></div>
<p><img src="Introduction-to-mixt_files/figure-html/unnamed-chunk-35-1.png" width="384"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#input-data-format">1. Input data format</a></li>
      <li><a href="#patient-linear-ordering-and-region-of-independence">2. Patient linear ordering and region of independence</a></li>
      <li><a href="#functional-enrichment">3. Functional enrichment</a></li>
      <li>
<a href="#association-between-gene-sets-and-clinical-variables">4. Association between gene sets and clinical variables</a><ul class="nav nav-pills nav-stacked">
<li><a href="#investigating-gene-overlap-between-gene-sets-across-tissues">5. Investigating gene overlap between gene sets across tissues</a></li>
      <li><a href="#matched-interactions-across-tissues">6. Matched Interactions Across Tissues</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Vanessa Dumeaux, Bjorn Fjukstad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
